src/test/java/net/openhft/chronicle/core/threads/TimerTest.java

```diff
@@ -0,0 +1,131 @@
+package net.openhft.chronicle.core.threads;
+
+import net.openhft.chronicle.core.time.SetTimeProvider;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.Closeable;
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class TimerTest {
+
+    public static final int INITIAL_DELAY_MS = 1_000;
+    public static final int PERIOD_MS = 2_000;
+    @Mock
+    private EventLoop eventLoop;
```

Comment:
There are no EventLoop implementations in Core so we have no choice but to mock here, also the tests read nicely with the mocked EventLoop.